tests/unit/app/endpoints/test_ols.py

```diff
@@ -173,6 +174,20 @@ def test_validate_question_valid_kw(llm_validate_question_mock, _load_config):
     assert llm_validate_question_mock.call_count == 0
 
 
+@patch(
+    "ols.src.query_helpers.question_validator.QuestionValidator.validate_question",
+    side_effect=PromptTooLongError("Prompt length 10000 exceeds LLM"),
+)
+def test_validate_question_too_long_query(llm_validate_question_mock, _load_config):
+    """Check the behaviour of validate_question function with too long query."""
+    conversation_id = suid.get_suid()
+    query = "Tell me about Kubernetes?" * 10000
+    llm_request = LLMRequest(query=query, conversation_id=conversation_id)
+    # PromptTooLongError should be caught and HTTPException needs to be raised
+    with pytest.raises(HTTPException, match="413: {'response': 'Prompt is too long'"):
+        ols.validate_question(conversation_id, llm_request)
```

Comment:
what is this really testing?  you setup a mock such that the exception is guaranteed to be raised.  So at best this is just testing that ols.validate_question is catching and reraising that exception?

Reply:
just this, it's UT, nothing on integration nor e2e level now

Reply:
for record, e2e waits for https://github.com/openshift/lightspeed-service/pull/820 and then https://github.com/openshift/lightspeed-service/pull/810 . when these two will be in, e2e should be created with real end-to-end behavior checking